trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/index.test.ts

```diff
@@ -212,11 +214,71 @@ describe('controllers/utilisation-reports/add-to-an-existing-payment', () => {
       expect((res._getRenderData() as AddToAnExistingPaymentViewModel).reportId).toEqual('123');
     });
 
+    it('should set the back link href', async () => {
+      // Arrange
+      const { req, res } = httpMocks.createMocks({
+        session: requestSession,
+        params: { reportId },
+        body: {
+          'feeRecordIds-1,22-reportedPaymentsCurrency-USD-status-TO_DO': 'on',
+          'feeRecordIds-333-reportedPaymentsCurrency-USD-status-TO_DO': 'on',
+        },
+      });
+      jest.mocked(api.getSelectedFeeRecordsDetailsWithAvailablePaymentGroups).mockResolvedValue({
+        ...aSelectedFeeRecordsDetails(),
+        feeRecords: [
+          {
+            ...aSelectedFeeRecordDetails(),
+            id: 1,
+            reportedFee: { amount: 2000, currency: 'EUR' },
+          },
+          {
+            ...aSelectedFeeRecordDetails(),
+            id: 22,
+            reportedFee: { amount: 3000, currency: 'EUR' },
+          },
+          {
+            ...aSelectedFeeRecordDetails(),
+            id: 333,
+            reportedFee: { amount: 100, currency: 'JPY' },
+          },
+        ],
```

Comment:
I can see very similar mocks repeated in 6x places prior to this PR, are we able to create some DRY test mocks?

Reply:
Reviewed this test again and noticed that we don't need to override `reportedFee` so this test has now been simplified to only modify the ID attribute on each fee record (as the test is dependent on these).

I've looked over the duplications you mentioned and I don't feel like they should be pulled out. The majority are setting up a mocked object to test data sort values are correct - the tests are directly dependent on the values that we override (the fee records reported fees) and the other attributes of the object are already pulled out into helper functions (e.g. `aSelectedFeeRecordDetails`). I think this approach is quite nice personally. Even so, now this test has been simplified I think this would be out of scope of this PR.